src/metabase/util.clj

```diff
@@ -62,8 +63,30 @@
                                                                                (pprint-to-str (sort (keys time/formatters)))))))))
 
 
+(defprotocol ISO8601
+  "Protocol for converting objects to ISO8601 formatted strings."
+  (->ISO8601DateTime ^String [this timezone-id]
```

Comment:
Since `ISO8601DateTime` isn't a class or class-like object this method should probably be named something like 
`->iso-8601-datetime` instead. Otherwise it's a bit misleading